crates/hyperswitch_domain_models/src/payments.rs

```diff
@@ -540,3 +542,14 @@ where
     /// This will depend on the payment status and the force sync flag in the request
     pub should_sync_with_connector: bool,
 }
+
+#[cfg(feature = "v2")]
+#[derive(Clone)]
+pub struct PaymentUpdateData<F>
+where
+    F: Clone,
+{
+    pub flow: PhantomData<F>,
+    pub payment_intent: PaymentIntent,
+    pub payment_intent_update: PaymentIntentUpdate,
```

Comment:
this object is not required, in PaymentsState. Is this necessary to be placed in state?